lib/markdown.js

```diff
@@ -1,35 +1,45 @@
 /* jshint node:true */
 "use strict";
 
+var rHeader = /^\s*(\#+|={3,}|-{3,})/;
+var rBadge = /^\[!\[[^\]]+\]\([^)]+\)\]\([^)]+\)$/;
+
 function markdownBadger(content, imgUrl, linkUrl, altText) {
   var badgeContent = getBadgeMarkdown(imgUrl, linkUrl, altText);
 
   var lines = content.split(/\n/);
-  var idealLine = findIdealLineForInsert(lines) || 0;
-
-  lines.splice(idealLine, 0, badgeContent);
+  var idealLine = findLastBadgeLine(lines);
+  if(idealLine >= 0) {
+    lines.splice(idealLine + 1, 0, badgeContent);
+  }else{
+    idealLine = findHeaderLine(lines);
+    // if the next is not an empty line, add another empty line
+    if(lines[idealLine + 1]) badgeContent += '\n';
+    lines.splice(idealLine + 1, 0, '\n' + badgeContent);
+  }
 
   return lines.join('\n');
 }
 
-function findIdealLineForInsert(lines) {
-  if(lines.length === 0) return 0;
+function findHeaderLine(lines) {
   var i = 0;
-  var seenHeader = false;
 
   for(;i < lines.length;i++) {
-    if(/^\s*(\#+|={3,}|-{3,})/.test(lines[i])) {
-      seenHeader = true;
-    } else {
-      if(seenHeader) break;
-    }
+    if(rHeader.test(lines[i])) return i;
   }
+  return -1;
+}
+function findLastBadgeLine(lines) {
+  var i = lines.length - 1;
 
-  return i;
+  for(;i >= 0;i--) {
```

Comment:
Prefer `for(var i = lines.length - 1; i >= 0; i--) {`

Reply:
Agreed. But I followed [the original style](https://github.com/gitterHQ/readme-badger/blob/master/lib/markdown.js#L20). Is it because of the last [return statement](https://github.com/gitterHQ/readme-badger/blob/master/lib/markdown.js#L28) includes `i`?

Reply:
Ahh, I didn't notice it was legacy. It's fine to leave it as-is 🤷‍♀️